CHANGELOG.md

```diff
@@ -3,6 +3,73 @@
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.1.0
+
+_Nov 2, 2021_
+
+A big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:
+
+### `@mui/x-data-grid@v5.1.0` / `@mui/x-data-grid-pro@v5.1.0`
+
+- 🚀 Introduce the [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data) feature (#2725) @flaviendelangle
+
+  <img src="https://user-images.githubusercontent.com/42154031/144259216-fc4f90ac-4d8b-4253-bc95-009204349a4c.gif" width="854" height="453" />
+
+- 💅 Add support for `sx` prop in the DataGrid and DataGridPro (#3281) @m4theushw
+- 🔦 Improve focus management in the filter panel (#3004) @alexfauquette
+- 🌎 Translation updates for many locales
+
+  If you are using DataGrid or DataGridPro in another language, check [this issue](https://github.com/mui-org/material-ui-x/issues/3211) to discover which translations are missing.
+
+- 📚 Documentation improvements
+- 🐞 Bugfixes
+
+#### Changes
+
+- [DataGridPro] Add tree data (#2725) @flaviendelangle
+- [DataGridPro] Improve license file (#3278) @oliviertassinari
+- [DataGridPro] Add types for event publishers and listeners (#3022) @flaviendelangle
+- [DataGrid] Add support for `sx` prop (#3281) @m4theushw
+- [DataGrid] Do not debounce the initial resizing of the grid (#3213) @flaviendelangle
+- [DataGrid] Fix usage of dynamic columns (#3204) @flaviendelangle
+- [DataGrid] Move focus when selecting option with <kbd>Enter</kbd> in the `singleSelect` (#3220) @m4theushw
+- [DataGrid] Focus on the last value input when a filter is added or removed (#3004) @alexfauquette
+- [DataGrid] Prepare the tree structure for grouping sorting / filtering (#3301) @flaviendelangle
+- [DataGrid] Rework keyboard navigation (#3193) @flaviendelangle
+- [DataGrid] Set minimum dimensions to `GridOverlay` when no row is provided (#3261) @flaviendelangle
+- [DataGrid] Improve German (deDE) locale (#3271, #3230, #3293) @sebastianfrey
+- [DataGrid] Improve Hebrew (heIL) locale (#3294) @ColdAtNight
+- [DataGrid] Improve Russian (ruRU) locale (#3290, #3288) @Alim-El
+- [DataGrid] Improve Korean (koKR) locale (#3232, #3273) @zzossig
+- [DataGrid] Improve Greek (elGR) locale (#3169) @clytras
+
+### Core
+
+- [core] Add script to sync translation files (#3201) @m4theushw
+- [core] Create dedicated `InputComponent` for `singleSelect` and `date` columns #3227 @alexfauquette
+- [core] Fix `EventManager` to not run listeners removed after registration #3206 @flaviendelangle
+- [core] Group Renovate updates (#3263) @flaviendelangle
+- [test] Improve testing of the keyboard navigation (#3187) @flaviendelangle
+- [core] Reflect the change of default branch (#3235) @oliviertassinari
+- [core] Replace @mui/core with @mui/base (#3217) @m4theushw
+- [core] Split docs generation script (#3189) @flaviendelangle
+- [core] Update monorepo (#3303) @m4theushw
+- [test] Force effect to run on location change (#3283) @m4theushw
+
+### Docs
```

Comment:
The `core` has been above `docs` for a long time but I agree that the oppose makes more sense

Reply:
Why "docs" changes are more important than "core"? Different from the core, we use this category even for changes in the DataGrid component but which are not user-facing. In the core, it seems to be used only for changes in the infrastructure, which makes sense to keep them at the very end.

Reply:
> Why "docs" changes are more important than "core"

I think because docs is as important as the feature if not more, while core is infrastructure that happens behind the scene, e.g. I assume people don't care we have a new performance benchmark while they care if we add a guide about server side data loading.

Reply:
In this particular release, most of the "core" changes are true core changes, only modifying infrastructure. Looking back, you're gonna see a lot of internal changes labelled as "core". I'm favor of always labelling them with the component name they belong to, e.g. "DataGrid", even for internal changes. Doing that we follow the same convention adopted by the core repo, while it's also more scalable when more components are added.

https://github.com/mui-org/material-ui/blob/81f445cfeb4bf0c231818e80b699f3102736548c/CHANGELOG.md?plain=1#L4232

Reply:
> [SwipeableDrawer] Refactor internals (#23944)

This one is interesting, I think that "[core] Refactor SwipeableDrawer internals (#23944)" would have been more accurate. Developers don't need to know about #23944, it's mostly noise.